pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/ActivityInstanceDao.java

```diff
@@ -270,7 +270,7 @@ int bulkUpdateReadOnlyByActivityIds(
             @BindList(value = "activityIds", onEmpty = EmptyHandling.NULL) Set<Long> activityIds);
 
     @SqlUpdate("update activity_instance set is_hidden = :isHidden"
-            + "  where participant_id = :participantId and study_activity_id in (<activityIds>)")
+            + "  where participant_id = :participantId and study_activity_id in (<activityIds>) order by activity_instance_id")
```

Comment:
Trying to understand how the order by in this update going to help. 
All rows that need update (or updated) are locked any way until the transaction is committed, right ? Order by will make the query take more time to complete but I believe updates happen once rows are selected any way. 
It doesn't seem to hurt though.

Reply:
If one activity_instance row  is locked by from problem#2 conn, isn't this problem#1 conn going to just wait by locking other activity_instance rows!